### ISSUE TITLE
Not sure if model_checkpoint is able to load different checkpoints

### ISSUE THREAD

**Issue by mjyb16** [AlexandreAdam/torch_score_models#4]
I am unsure how to ensure I am loading the correct checkpoint, because I am getting inconsistent results. Note: this is the pip installed version (0.5.5)

**Reproduce:** 
`score1 = ScoreModel(model_checkpoint = 10.0, seed = 0, checkpoints_directory="/home/mjybarth/scratch/cats_1_checkpoints")`
Loaded checkpoint 0 of cats_1_checkpoints

`score1 = ScoreModel(model_checkpoint = 9.0, seed = 0, checkpoints_directory="/home/mjybarth/scratch/cats_1_checkpoints")`
Loaded checkpoint 0 of cats_1_checkpoints

**Expected behavior:**
Code, taken from the package itself, parameters set by hand:
```
model_checkpoint = 9.0
checkpoints_directory="/home/mjybarth/scratch/cats_1_checkpoints"
paths = glob.glob(os.path.join(checkpoints_directory, "checkpoint*.pt"))
opt_paths = glob.glob(os.path.join(checkpoints_directory, "optimizer*.pt"))
checkpoint_indices = [int(re.findall('[0-9]+', os.path.split(path)[-1])[-1]) for path in paths]
scores = [float(re.findall('([0-9]{1}.[0-9]+e[+-][0-9]{2})', os.path.split(path)[-1])[-1]) for path in paths]
if checkpoint_indices:
    if model_checkpoint is not None:
        checkpoint_path = paths[checkpoint_indices.index(model_checkpoint)]
        print(f"Loaded checkpoint {model_checkpoint} of cats_1_checkpoints")
        print(checkpoint_path)
        latest_checkpoint = model_checkpoint
        
else:
    max_checkpoint_index = np.argmax(checkpoint_indices)
    checkpoint_path = paths[max_checkpoint_index]
    opt_path = opt_paths[max_checkpoint_index]
    print(f"Loaded checkpoint {checkpoint_indices[max_checkpoint_index]} of cats_1_checkpoints")
    latest_checkpoint = checkpoint_indices[max_checkpoint_index]
```
Loaded checkpoint 9.0 of cats_1_checkpoints
/home/mjybarth/scratch/cats_1_checkpoints/checkpoint_5.5174e+02_009.pt

**Comment by AlexandreAdam** [comment]
Good catch. I removed this functionality in the ScoreModel `load_architecture` function. The latest checkpoint is always loaded. In principle, the `fit` method can work from earlier checkpoints with the `model_checkpoint` argument, but in practice I've never really used that functionality since early stopping is not useful when training score models. 

I can add this functionality back in the `load_architecture` function. The default behavior will be to load the latest checkpoint present in the checkpoints directory.

**Comment by AlexandreAdam** [comment]
This issue should be fixed in version 0.5.8. Let me know if you still have trouble.
